### PR TITLE
OGM-1458 Change mapping of byte type field to ProtoBuf int32

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/OGM-1458

See also: https://github.com/infinispan/infinispan/pull/6191

### CODE REVIEW

File: core/src/test/java/org/hibernate/ogm/backendtck/queries/parameters/QueryWithParametersTest.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 	public PackagingRule packaging = new PackagingRule( "persistencexml/ogm.xml", Movie.class );
 
 	@Test
-	@SkipByGridDialect(value = { INFINISPAN_REMOTE }, comment = "Query on Byte field are currently not supported")
+	@SkipByGridDialect(value = { INFINISPAN_REMOTE }, comment = "Query on --starting with 'v'-- field are currently not supported")
```
Comment (gsmet) [inline core/src/test/java/org/hibernate/ogm/backendtck/queries/parameters/QueryWithParametersTest.java]: This looks weird. What does it mean?

Comment (anistor) [reply core/src/test/java/org/hibernate/ogm/backendtck/queries/parameters/QueryWithParametersTest.java]: A strange reference to ISPN-7863 , maybe :))))

Comment (gsmet) [reply core/src/test/java/org/hibernate/ogm/backendtck/queries/parameters/QueryWithParametersTest.java]: Hum, OK. AFAICS, it's fixed in 9.3.4.Final so I don't think it's worth it to reference this ISPN bug in our code.

Comment (fax4ever) [reply core/src/test/java/org/hibernate/ogm/backendtck/queries/parameters/QueryWithParametersTest.java]: Thank you @gsmet and @anistor. I will change the comment here.

File: infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/impl/protobuf/ByteProtofieldAccessor.java [hunk 2]
```diff
@@ -11,29 +11,24 @@
 
 /**
  * There is no explicit "byte" support in Protostream.
- * Decided to encode it as a byte array type; on reading a longer sequence it will be truncated as we only read the first byte.
+ * Decided to encode it as a int type; on reading an int will be truncated as a byte,
+ * according to the endian behavior adopted by the {@link Integer#byteValue()} method
```
Comment (Sanne) [inline infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/impl/protobuf/ByteProtofieldAccessor.java]: What _endian behaviour_ ?

We need to make sure the encoding is the same independently from the specific platform this is running on.

I assumed the byteValue() method provided a stable specification, but either I'm wrong or this comment is overly concerning :)

Comment (fax4ever) [reply infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/impl/protobuf/ByteProtofieldAccessor.java]: this comment is definitely overly concerning :) I think we can rely on a consistency of Integer#byteValue(). I will change it. Thank you :)

**Comment by fax4ever** [comment]
@Sanne I changed the comment here :)

**Comment by Sanne** [comment]
merged!

**Comment by fax4ever** [comment]
Thank you @Sanne @anistor @gsmet for the advice here

**Pull request merged**